doc/features/pre-existing-tenant.md

```diff
@@ -0,0 +1,59 @@
+
+## 2. Motivation
+
+The pre-existing tenant feature be able to process the tenant field in ACI config of the input yaml file to provision the Kubernetes cluster under the given tenant. This includes creating create application profile, EPGs, BDs and required contracts under the existing tenant. If this tenant isn’t already created, our tool will create it, but not delete it in the un-provision step. All other APIC objects created for this cluster being un-provisioned will be deleted. The only case where the tool will delete a tenant is when the tenant was created by our tool (identified by a unique tag), and has no other clusters provisioned under it.
+
+## 3. Mechanism
+
+acc-provision support 2 mode of handling tenant:
+1. It can create a dedicated tenant for the cluster
+2. A shared/pre-existing tenant when the tenant name is specified
+
+Depending on the mode the following behaviours are expected when the config is de-provisioned by using the `-d` option -
+
+1. The whole tenant is deleted
+2. Only the config added by acc-provision is deleted
+
+In case 1 scenario, initially if you run acc-provision with the -d option the tenant is not deleted if there is at least 1 application profile. However the tenant gets deleted in all other cases. e.g with pre-existing: VRFs, BDs, L3OUT and now if the "tenant name" is not specified in the acc provision input file the whole tenant should be deleted irrespective of number of application profile.
+
+In case 2 scenario, when it's pre-existing-tenant i.e when the tenant name is specified in the acc provision input file in aci_config section, the manually created tenant on the APIC before provisioning is not being deleted when unprovisioning i.e -d option but only the config added by provisioning i.e -a is deleted.
+
+## 4. Lifecycle
+
+A Kubernetes cluster can be provisioned under a tenant by adding this config in the input yaml file -
+
+```yaml
+aci_config:
+  tenant:
+    name: pre_existing_tenant
+```
+
+acc-provision will create this tenant if not present but will never delete this tenant. Additionally, this instructs acc-provision to create application profile, EPGs, BDs and required contracts under the existing tenant. The “system_id” field provided in the acc-provision input file is used to construct names of these APIC objects. For example, application profile for the cluster with ID “cluster1” will be called “aci-containers-cluster1” and contract for DNS will be called “aci-containers-cluster1-dns”. This naming convention change is for all kinds of clusters provisioned by our tool, not just pre-existing tenant ones. Aci-prefix is only tied to systemid but not to a tenant. If a null or blank value is provided in the tenant name field, its value will default to the system_id provided in the input config file. To summarize - if tenant is manually created on the APIC before provisioning have tenant name in acc provision input file under aci_config section.
```

Comment:
Warning: When selecting the tenant and system ID name keep in mind that if your Nodes are deployed as VMs on VmWare the maximum port-group name is 80 character and will be composed by the concatenation of the Tenant + App Profile + EPG Name

Maybe this could fit also in a different place but I think is worth mentioning